src/contracts/test/GPv2EncodingTestInterface.sol

```diff
@@ -73,6 +73,34 @@ contract GPv2EncodingTestInterface {
         gas_ = gas_ - gasleft();
     }
 
+    function decodeInteractionsTest(
+        bytes calldata encodedInteractions,
+        uint256 expectedInteractionCount
+    )
+        external
+        view
+        returns (GPv2Encoding.Interaction[] memory interactions, uint256 gas_)
+    {
+        interactions = new GPv2Encoding.Interaction[](expectedInteractionCount);
+
+        uint256 interactionCount = 0;
+        while (encodedInteractions.length != 0) {
```

Comment:
ubernit: maybe call `remainingInteractions` given that it's mutated within the function?